### PR TITLE
Troop controller

### PR DESCRIPTION
**Pull request by DanoFecko**
*TroopController

### CODE REVIEW

**Comment by GreenManSK** [comment]
You are missing all the templates and checks for user roles

**Comment by DanoFecko** [comment]
Hi @GreenManSK, I add all necessary files, that have been missing

File: dnd-troops-web/src/main/resources/Texts.properties [hunk 1]
```diff
@@ -67,6 +67,22 @@ role.search=Search
...
+troop.delete.problem=Problem while deleting troop
+troop.delete.success=Troop successfully deleted
+troop.create=Create Troop
+troop.create.success=Troop successfully created
+troop.update=Update role
+troop.update.success=Troop updated successfully
+troop.search=Search
+troop.search.problem=There has been problem during search
+troop.search.not.found=No troop found
+
```
Comment (GreenManSK) [inline dnd-troops-web/src/main/resources/Texts.properties]: Missing troop.data.problem

File: dnd-troops-web/src/main/resources/Texts.properties [hunk 2]
```diff
@@ -67,6 +67,22 @@ role.search=Search
 role.search.problem=Problem with searching role
 role.search.not.found=There is no role with given name
 
+#troop
+troop.all.title=Troops
+troop.all.add=AddRole
```
Comment (GreenManSK) [inline dnd-troops-web/src/main/resources/Texts.properties]: Nope

File: dnd-troops-web/src/main/java/cz/muni/fi/pa165/w2018/dndtroops/web/controllers/TroopController.java [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+@RequestMapping(value = {WebUris.URL_TROOP})
+public class TroopController extends BaseController {
+
+	private final static Logger log = LoggerFactory.getLogger(TroopController.class);
+
+	@Autowired
+	private TroopFacade troopFacade;
+
+	@RequestMapping(value = "/search/name", method = RequestMethod.POST)
+	public String searchName(
```
Comment (GreenManSK) [inline dnd-troops-web/src/main/java/cz/muni/fi/pa165/w2018/dndtroops/web/controllers/TroopController.java]: Why this exists when you have findByName and findByMission handlers

File: dnd-troops-web/src/main/java/cz/muni/fi/pa165/w2018/dndtroops/web/controllers/TroopController.java [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+	 *
+	 * @param name String
+	 */
+	@RequestMapping(value = {"/findByName/{name}"}, method = RequestMethod.GET)
+	public String findByName(Model model, @PathVariable String name) {
+		log.debug("findByName(" + name + ")");
+		List<TroopDTO> troops = troopFacade.getAllByName(name);
+		model.addAttribute("troops", troops);
+
+		return "/troop/findByName";
```
Comment (GreenManSK) [inline dnd-troops-web/src/main/java/cz/muni/fi/pa165/w2018/dndtroops/web/controllers/TroopController.java]: Missing jsp

File: dnd-troops-web/src/main/java/cz/muni/fi/pa165/w2018/dndtroops/web/controllers/TroopController.java [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+	 *
+	 * @param mission String
+	 */
+	@RequestMapping(value = {"/findByMission/{mission}"}, method = RequestMethod.GET)
+	public String findByMission(Model model, @PathVariable String mission) {
+		log.debug("findByMission(" + mission + ")");
+		List<TroopDTO> troops = troopFacade.getAllByMission(mission);
+		model.addAttribute("troops", troops);
+
+		return "/troop/findByMission";
```
Comment (GreenManSK) [inline dnd-troops-web/src/main/java/cz/muni/fi/pa165/w2018/dndtroops/web/controllers/TroopController.java]: Missing jsp

File: dnd-troops-web/src/main/webapp/WEB-INF/jsp/troop/all.jsp [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<%@ taglib prefix="s" uri="http://www.springframework.org/tags" %>
+<%@ taglib prefix="form" uri="http://www.springframework.org/tags/form" %>
+
+<fmt:message var="title" key="troop.all.title"/>
+<my:admintemplate title="${title}">
+    <jsp:attribute name="body">
+        <h1><fmt:message key="troop.all.title"/></h1>
+
+        <form:form method="post" action="${pageContext.request.contextPath}/troop/search/name" modelAttribute="search" id="searchTroopName">
```
Comment (GreenManSK) [inline dnd-troops-web/src/main/webapp/WEB-INF/jsp/troop/all.jsp]: User should be able to:
Find by name
Find by mission

Comment (DanoFecko) [reply dnd-troops-web/src/main/webapp/WEB-INF/jsp/troop/all.jsp]: image

which one is right?

Comment (GreenManSK) [reply dnd-troops-web/src/main/webapp/WEB-INF/jsp/troop/all.jsp]: Both, there should be two different ways of searching for troop

**Pull request merged**